src/behat/Configuration/SnmpTrapsConfigurationPage.php

```diff
@@ -0,0 +1,305 @@
+            self::TAB_ADVANCED
+        ),
+        'execution_type' => array(
+            'radio',
+            'input[name="traps_exec_interval_type[traps_exec_interval_type]"]',
+             self::TAB_ADVANCED
+        ),
+        'execution_method' => array(
+            'radio',
+            'input[name="traps_exec_method[traps_exec_method]"]',
+            self::TAB_ADVANCED
+        ),
+        'check_downtime' => array(
+            'radio',
+            'input[name="traps_downtime[traps_downtime]"]',
+            self::TAB_ADVANCED
+        ),
+        'output_transform' => array(
+            'text',
+            'input[name="traps_output_transform"]',
+            self::TAB_ADVANCED
+        ),
+        'custom_code' => array(
+            'text',
+            'textarea[name="traps_customcode"]',
+            self::TAB_ADVANCED
+        )
+    );
+
+    /**
+     * @var string
+     */
+    protected $listingClass = '\Centreon\Test\Behat\Configuration\SnmpTrapsConfigurationListingPage';
+
+    /**
+     *  Navigate to and/or check that we are on a SNMP trap configuration
+     *  page.
+     *
+     *  @param $context  Centreon context.
+     *  @param bool $visit    True to navigate to a blank configuration page.
+     */
+    public function __construct($context, $visit = TRUE)
+    {
+        // Visit page.
+        $this->context = $context;
+        if ($visit) {
+            $this->context->visit('main.php?p=61701&o=a#');
+        }
+
+        // Check that page is valid for this class.
+        $mythis = $this;
+        $this->context->spin(
+            function ($context) use ($mythis) {
+                return $mythis->isPageValid();
+            },
+            'Current page does not match class ' . __CLASS__
+        );
+    }
+
+    /**
+     *get rule array
+     *
+     *@return rule array
+     */
+    protected function getRuleArray($i)
+    {
+        $rule = $this->context->getSession()->getPage()->findField('rule[' . $i . ']');
+        if (is_null($rule)) {
+            return null;
+        }
+        $ruleArray['string'] = $rule->getValue();
+        $ruleArray['regexp'] = $this->context->getSession()->getPage()->findField('regexp[' . $i . ']')->getValue();
+        if (!is_null($this->context->getSession()->getPage()->findField('#rulestatus_' . $i))) {
+            $ruleArray['status'] = $this->context->assertFind('css', 'select#rulestatus_' . $i . ' option:selected')->getText();
+        }
+        if (!is_null($this->context->getSession()->getPage()->findField('#ruleseverity_' . $i))) {
+            $ruleArray['severity'] = $this->context->assertFind('css', 'select#ruleseverity_' . $i . ' option:selected')->getText();
+        }
+        return $ruleArray;
+    }
+
+    /**
+     *get rule
+     *
+     *@return rule
+     */
+    protected function getRule()
+    {
+        $rule = array();
+        $i = 0;
+        while (true) {
+            if (is_null($this->getRuleArray($i))) {
```

Comment:
You can use findAll method to get all rules with one call
Then you can loop on it, and use assertFindIn method